src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java

```diff
@@ -0,0 +1,124 @@
+package io.cdap.plugin.bigquery.stepsdesign;
+
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BigQueryMultiTableValidation {
+  static Gson gson = new Gson();
+  public static boolean validateBQToBigQueryMultiTableInTwoTables(List<String> sourceTables) throws IOException, InterruptedException {
+    List<String> targetTables = getTableByName().stream().sorted().collect(Collectors.toList());
+    if (sourceTables.size() != targetTables.size()) {
+      throw new IllegalArgumentException("Number of source tables and target tables must be the same.");
+    }
+    for (int i = 0; i < sourceTables.size(); i++) {
+      String currentSourceTable = sourceTables.get(i);
+      String currentTargetTable = targetTables.get(i);
+      List<Object> bigQueryRows = new ArrayList<>();
+      getBigQueryTableData(currentSourceTable, bigQueryRows);
+      List<JsonObject> bigQueryResponse = new ArrayList<>();
+      for (Object row : bigQueryRows) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse.add(jsonData);
+      }
+      List<Object> bigQueryRows2 = new ArrayList<>();
+      getBigQueryTableData(currentTargetTable, bigQueryRows2);
+
+      List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+      for (Object row : bigQueryRows2) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse2.add(jsonData);
+      }
```

Comment:
This logic can be extracted to a separate function to reduce code repetition !

Reply:
done